packages/cheqd/tests/cheqd-data-integrity.e2e.test.ts

```diff
@@ -0,0 +1,273 @@
+import type { AnonCredsTestsAgent } from './helpers/cheqdAnonCredsSetup'
+import type { EventReplaySubject } from '../../core/tests'
+import type { InputDescriptorV2 } from '@sphereon/pex-models'
+
+import {
+  AutoAcceptCredential,
+  CredentialExchangeRecord,
+  CredentialState,
+  ProofState,
+  W3cCredential,
+  W3cCredentialService,
+  W3cCredentialSubject,
+} from '@credo-ts/core'
+
+import { InMemoryAnonCredsRegistry } from '../../anoncreds/tests/InMemoryAnonCredsRegistry'
+import { presentationDefinition } from '../../anoncreds/tests/fixtures/presentation-definition'
+import { createDidKidVerificationMethod } from '../../core/tests'
+import { waitForCredentialRecordSubject, waitForProofExchangeRecord } from '../../core/tests/helpers'
+
+import { setupAnonCredsTests } from './helpers/cheqdAnonCredsSetup'
+
+describe('anoncreds w3c data integrity e2e tests', () => {
+  let issuerId: string
+  let issuerAgent: AnonCredsTestsAgent
+  let holderAgent: AnonCredsTestsAgent
+  let credentialDefinitionId: string
+  let issuerHolderConnectionId: string
+  let holderIssuerConnectionId: string
+
+  let issuerReplay: EventReplaySubject
+  let holderReplay: EventReplaySubject
+
+  const inMemoryRegistry = new InMemoryAnonCredsRegistry()
+
+  afterEach(async () => {
+    await issuerAgent.shutdown()
+    await issuerAgent.wallet.delete()
+    await holderAgent.shutdown()
+    await holderAgent.wallet.delete()
+  })
+
+  test('cheqd issuance and verification flow starting from offer without revocation', async () => {
+    ;({
+      issuerAgent,
+      issuerReplay,
+      holderAgent,
+      holderReplay,
+      credentialDefinitionId,
+      issuerHolderConnectionId,
+      holderIssuerConnectionId,
+      issuerId,
+    } = await setupAnonCredsTests({
+      issuerName: 'Faber Agent Credentials v2',
+      holderName: 'Alice Agent Credentials v2',
+      attributeNames: ['id', 'name', 'height', 'age'],
+      registries: [inMemoryRegistry],
+    }))
+    await anonCredsFlowTest({
+      issuerId,
+      credentialDefinitionId,
+      issuerHolderConnectionId,
+      holderIssuerConnectionId,
+      issuerReplay,
+      holderReplay,
+      issuer: issuerAgent,
+      holder: holderAgent,
+    })
```

Comment:
as mentioned in previous review. If we only have on test there's no need to extract it into a method. Makes it harder to udnerstand the flow IMO